Modules/_datetimemodule.c

```diff
@@ -23,32 +23,85 @@
+    PyObject *us_per_hour;     // 1e6 * 3600 as Python int
+    PyObject *us_per_day;      // 1e6 * 3600 * 24 as Python int
+    PyObject *us_per_week;     // 1e6*3600*24*7 as Python int
+    PyObject *seconds_per_day; // 3600*24 as Python int
+
+    /* The interned UTC timezone instance */
+    PyObject *PyDateTime_TimeZone_UTC;
+
+    /* The interned Epoch datetime instance */
+    PyObject *PyDateTime_Epoch;
 
-#define PyTime_Check(op) PyObject_TypeCheck(op, &PyDateTime_TimeType)
-#define PyTime_CheckExact(op) Py_IS_TYPE(op, &PyDateTime_TimeType)
+    /* strptime import */
+    PyObject *strptime;
 
-#define PyDelta_Check(op) PyObject_TypeCheck(op, &PyDateTime_DeltaType)
-#define PyDelta_CheckExact(op) Py_IS_TYPE(op, &PyDateTime_DeltaType)
+    /* Types */
+    PyTypeObject *PyDateTime_DateTimeType;
+    PyTypeObject *PyDateTime_DateType;
+    PyTypeObject *PyDateTime_DeltaType;
+    PyTypeObject *PyDateTime_IsoCalendarDateType;
+    PyTypeObject *PyDateTime_TZInfoType;
+    PyTypeObject *PyDateTime_TimeType;
+    PyTypeObject *PyDateTime_TimeZoneType;
+} datetime_state;
 
-#define PyTZInfo_Check(op) PyObject_TypeCheck(op, &PyDateTime_TZInfoType)
-#define PyTZInfo_CheckExact(op) Py_IS_TYPE(op, &PyDateTime_TZInfoType)
+static inline datetime_state *
+get_module_state(PyObject *mod)
+{
+    void *state = PyModule_GetState(mod);
+    assert(state != NULL);
+    return (datetime_state *)state;
+}
 
-#define PyTimezone_Check(op) PyObject_TypeCheck(op, &PyDateTime_TimeZoneType)
+static struct PyModuleDef datetimemodule;
+
+static inline datetime_state *
+find_module_state_by_def(PyTypeObject *type)
+{
+    PyObject *mod = PyType_GetModuleByDef(type, &datetimemodule);
+    assert(mod != NULL);
+    return get_module_state(mod);
+}
+
+static inline datetime_state *
+find_state_left_or_right(PyObject *left, PyObject *right)
+{
+    PyObject *mod = PyType_GetModuleByDef(Py_TYPE(left), &datetimemodule);
+    if (mod == NULL) {
+        PyErr_Clear();
+        mod = PyType_GetModuleByDef(Py_TYPE(right), &datetimemodule);
+    }
+    assert(mod != NULL);
+    return get_module_state(mod);
+}
+
+#define find_module_state_by_def(obj) find_module_state_by_def((PyTypeObject *)(obj))
 
 /*[clinic input]
 module datetime
-class datetime.datetime "PyDateTime_DateTime *" "&PyDateTime_DateTimeType"
-class datetime.date "PyDateTime_Date *" "&PyDateTime_DateType"
-class datetime.IsoCalendarDate "PyDateTime_IsoCalendarDate *" "&PyDateTime_IsoCalendarDateType"
+class datetime.datetime "PyDateTime_DateTime *" "clinic_state()->PyDateTime_DateTimeType"
+class datetime.date "PyDateTime_Date *" "clinic_state()->PyDateTime_DateType"
+class datetime.IsoCalendarDate "PyDateTime_IsoCalendarDate *" "clinic_state()->PyDateTime_IsoCalendarDateType"
 [clinic start generated code]*/
-/*[clinic end generated code: output=da39a3ee5e6b4b0d input=81bec0fa19837f63]*/
+/*[clinic end generated code: output=da39a3ee5e6b4b0d input=8a333fc5cbf6a146]*/
 
+#define clinic_state() (fixme())
```

Comment:
There is no need to fix it since it's unused. Maybe just use abort() and add a comment to remained that it's unused?

Reply:
Yeah, or maybe just remove it.